las_geoh5/import_las.py

```diff
@@ -46,23 +105,34 @@ def get_depths(lasfile: lasio.LASFile) -> dict[str, np.ndarray]:
     return out
 
 
-def get_collar(lasfile: lasio.LASFile) -> list | None:
+def get_collar(
+    lasfile: lasio.LASFile, translator: LASTranslator | None = None
+) -> list | None:
     """
     Returns collar data from las file or None if data missing.
 
     :param lasfile: Las file object.
 
     :return: Collar data.
     """
+
+    if translator is None:
+        translator = LASTranslator()
+
     collar = []
-    for attr in ["X", "Y", "ELEV"]:
-        if hasattr(lasfile.well, attr):
-            collar.append(lasfile.well[attr].value)
+    for field in ["collar_x", "collar_y", "collar_z"]:
+        try:
+            collar.append(translator.retrieve(field, lasfile))
+        except KeyError:
+            warnings.warn(
+                f"{field.replace('_', ' ').capitalize()} field "
+                f"'{getattr(translator, field)}' not found in las file."

```

Comment:
We were thinking providing the list of fields available in file to help the user correct the inputs.

Reply:
done